From 8fed927ead535c0ade8a1343abdad99f5eeaf4f3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:14:14 -0400
Subject: [powerpc] powernv: Fix permissions on sysparam sysfs entries

Message-id: <1410545655-205645-626-git-send-email-dzickus@redhat.com>
Patchwork-id: 94411
O-Subject: [RHEL7 PATCH 625/626] powernv: Fix permissions on sysparam sysfs entries
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1bd098903fda069cb96fe8b5cb4595b46c683385
Author: Anton Blanchard <anton@samba.org>
Date:   Sat Jun 7 11:29:01 2014 +1000

    powernv: Fix permissions on sysparam sysfs entries

    Everyone can write to these files, which is not what we want.

    Cc: stable@vger.kernel.org # 3.15
    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/opal-sysparam.c b/arch/powerpc/platforms/powernv/opal-sysparam.c
index d202f9b..9d1acf2 100644
--- a/arch/powerpc/platforms/powernv/opal-sysparam.c
+++ b/arch/powerpc/platforms/powernv/opal-sysparam.c
@@ -260,10 +260,10 @@ void __init opal_sys_param_init(void)
    attr[i].kobj_attr.attr.mode = S_IRUGO;
    break;
   case OPAL_SYSPARAM_WRITE:
-   attr[i].kobj_attr.attr.mode = S_IWUGO;
+   attr[i].kobj_attr.attr.mode = S_IWUSR;
    break;
   case OPAL_SYSPARAM_RW:
-   attr[i].kobj_attr.attr.mode = S_IRUGO | S_IWUGO;
+   attr[i].kobj_attr.attr.mode = S_IRUGO | S_IWUSR;
    break;
   default:
    break;
-- 
1.7.1